jiant/models.py

```diff
@@ -152,8 +152,13 @@ def build_sent_encoder(args, vocab, d_emb, tasks, embedder, cove_layer):
     elif any(isinstance(task, LanguageModelingTask) for task in tasks) or args.sent_enc == "bilm":
         assert_for_log(args.sent_enc in ["rnn", "bilm"], "Only RNNLM supported!")
         assert_for_log(
-            args.input_module != "elmo" and not args.input_module.startswith("bert"),
-            "LM with full ELMo and BERT not supported",
+            not (
+                args.input_module == "elmo"
+                or args.input_module.startswith("bert")
+                or args.input_module.startswith("xlnet")
+            ),
+            f"input_module = {args.input_module} is not supported for lanugage modeling due to"
```

Comment:
It wasn't immediately obvious what "lookahead" referred to here - maybe "... uses bidirectional context. You should not use bidirectional encoders for language modeling, because they break left-to-right dependence."

(Also, is this the only place this is enforced? Might be worth a comment that you can disable this by removing this assert, in case someone wants to do experiments like https://arxiv.org/abs/1901.05287?)

Reply:
I don't think we enforce this anywhere else, but the Goldberg setup is also fairly unusual and delicate experimental, so I don't want to give the impression that it's properly supported until we've tested it. I revised the comment.